src/lib/router.ts

```diff
@@ -30,4 +32,69 @@ router.post('/matches', async (req, res, next) => {
   }
 });
 
+type Highlight = {
+  timestamp: number;
+  type: string;
+  videoSrc: string;
+};
+
+router.post('/matches/:id/highlights', async (req, res, next) => {
+  const id = req.params.id;
+  const { timestamp, type, videoSrc } = req.body;
+
+  if (!isValid(id)) {
+    res.status(400).send('Invalid match id');
+    return;
+  }
+
+  if (
+    typeof timestamp !== 'number' ||
+    typeof type !== 'string' ||
+    typeof videoSrc !== 'string'
+  ) {
+    res.status(400).send('Invalid payload');
+    return;
+  }
+
+  const newHighlight: Highlight = {
+    timestamp,
+    type,
+    videoSrc,
+  };
+
+  try {
+    const updated = await getMatchesCollection().updateOne(
+      { _id: new ObjectId(id) },
+      { $push: { highlights: newHighlight } }
+    );
+
+    if (!updated.acknowledged || updated.modifiedCount === 0) {
+      res.status(404).send('match identifier incorrect');
+      return;
+    }
+
+    const result = await getMatchesCollection().findOne({
+      _id: new ObjectId(id),
+    });
+    res.status(200).send(result);
+  } catch (error) {
+    next(error);
+  }
+});
+
+router.get('/matches', async (_, res, next) => {
```

Comment:
It's totally fine to use `_` here, still, I prefer `_req` or `_request` to understand, which parameter is not used.

Reply:
thanks, I don`t know that this works too